src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1330,7 +1330,19 @@ protected override void InvokeDefaultAction(string path)
                 invokeProcess.StartInfo.Arguments = path;
                 invokeProcess.Start();
 #elif CORECLR
-                throw new PlatformNotSupportedException();
+                try
+                {
+                    // Try Process.Start first. This works for executables even on headless SKUs.
+                    invokeProcess.StartInfo.FileName = path;
+                    invokeProcess.Start();
+                }
+                catch (Win32Exception)
+                {
+                    // If it's headless SKUs, rethrow.
+                    if (Platform.IsNanoServer || Platform.IsIoT) { throw; }
+                    // If it's full Windows, then try ShellExecute.
+                    ShellExecuteHelper.Start(invokeProcess.StartInfo);
```

Comment:
Is there a possibility that this call will throw? If it does, how are you handling it?